### PR TITLE
fix: sso login

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- fixes sso type configuration

#### Description of how you validated changes
- manual tests

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/system-config-manager.ts [hunk 1]
```diff
@@ -95,17 +95,19 @@ export const getProfiledAwsConfig = async (
       // need to force AWS_SDK_LOAD_CONFIG to a truthy value to force ProcessCredentials to prefer the credential process in ~/.aws/config instead of ~/.aws/credentials
       const sdkLoadConfigOriginal = process.env.AWS_SDK_LOAD_CONFIG;
       process.env.AWS_SDK_LOAD_CONFIG = '1';
-      const credentialProcess = new ProcessCredentials({ profile: profileName });
-      await credentialProcess.getPromise();
-      process.env.AWS_SDK_LOAD_CONFIG = sdkLoadConfigOriginal;
+      const chain = new CredentialProviderChain();
+      const processProvider = () => new ProcessCredentials({ profile: profileName, filename: configFilePath });
```
Comment (edwardfoyle) [inline packages/amplify-provider-awscloudformation/src/system-config-manager.ts:99]: Why is `filename: configFilePath` necessary?

File: packages/amplify-provider-awscloudformation/src/system-config-manager.ts [hunk 2]
```diff
@@ -95,17 +95,19 @@ export const getProfiledAwsConfig = async (
       // need to force AWS_SDK_LOAD_CONFIG to a truthy value to force ProcessCredentials to prefer the credential process in ~/.aws/config instead of ~/.aws/credentials
       const sdkLoadConfigOriginal = process.env.AWS_SDK_LOAD_CONFIG;
       process.env.AWS_SDK_LOAD_CONFIG = '1';
-      const credentialProcess = new ProcessCredentials({ profile: profileName });
-      await credentialProcess.getPromise();
-      process.env.AWS_SDK_LOAD_CONFIG = sdkLoadConfigOriginal;
+      const chain = new CredentialProviderChain();
+      const processProvider = () => new ProcessCredentials({ profile: profileName });
+      chain.providers.push(processProvider);
```
Comment (sobolk) [inline packages/amplify-provider-awscloudformation/src/system-config-manager.ts:100]: I'm guessing that ProcessCredentials provider lands at the end of the list.
Does this mean that additional providers in the chain will now take precedence ?
If yes then should we put ProcessCredentials at the beginning to maintain backwards compatibility?

Comment (lazpavel) [reply packages/amplify-provider-awscloudformation/src/system-config-manager.ts:100]: this is mostly a revert for a regression introduced by this change [here](https://github.com/aws-amplify/amplify-cli/pull/11821/files#diff-81c461c201c6a399d242fcf6409027b42824db979bfd47388f25668e0dd3bb60R95-R104), after discussing with Ed, the change that needs to stay is the process env variable logic

Comment (sobolk) [reply packages/amplify-provider-awscloudformation/src/system-config-manager.ts:100]: okay. Looks good then.

**Review by sobolk — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**